packages/cli/src/commands/governance/approve.ts

```diff
@@ -11,37 +11,47 @@ export default class Approve extends BaseCommand {
     ...BaseCommand.flags,
     proposalID: flags.string({ required: true, description: 'UUID of proposal to approve' }),
     from: Flags.address({ required: true, description: "Approver's address" }),
+    useMultiSig: flags.boolean({
+      required: false,
+      description: 'True means the request will be sent through multisig.',
+    }),
   }
 
   static examples = ['approve --proposalID 99 --from 0x5409ed021d9299bf6814279a6a1411a7e866a631']
 
   async run() {
     const res = this.parse(Approve)
     const account = res.flags.from
+    const useMultiSig = res.flags.useMultiSig
     const id = res.flags.proposalID
     this.kit.defaultAccount = account
     const governance = await this.kit.contracts.getGovernance()
-    const governanceApproverMultiSig = await this.kit.contracts.getMultiSig(
-      await governance.getApprover()
-    )
+    const multiSigAddress = await governance.getApprover()
```

Comment:
It seems weird to me to do this if `useMultiSig` is false

Reply:
ya it was a a tradeoff for overcomplexity, readability and avoiding duplicate logic.  i didn't want to be throwing if(useMultiSig) tertiaries on every statement.  and since this address is reused, it would be less efficient to re-call `await governance.getApprover` every time the address is needed.

i could obviously have a giant if/else block for the multisig vs non-multisig use cases, but then there is a lot of duplicate logic that is difficult to parse out into a separate method (eg. inserting an additional check in the checkBuilder).

if i don't explicitly instantiate the `governanceApproverMultiSig` when `useMultiSig` is false, then it requires in later logic needing to check if this variables is undefined prior to every use.

open to your further thoughts if you feel strongly about this.

Reply:
I think a general philosophy for contractkit is deferring as much as possible to JS because of the latency/load on the blockchain client (esp if the node is remote)

Reply:
ok -- i've updated to conditionally make the blockchain client request if multisig address is needed to be retrieved